internal/config/config.go

```diff
@@ -218,13 +216,41 @@ func (c *Config) NewMetalClient() *packngo.Client {
 	baseURL, _ := url.Parse(c.BaseURL)
 	baseURL.Path = path.Join(baseURL.Path, metalBasePath) + "/"
 	client, _ := packngo.NewClientWithBaseURL(consumerToken, c.AuthToken, standardClient, baseURL.String())
-	client.UserAgent = c.fullUserAgent(client.UserAgent)
+	client.UserAgent = c.tfSdkUserAgent(client.UserAgent)
 	c.metalUserAgent = client.UserAgent
 	return client
 }
 
-// NewMetalGoClient returns a new metal-go client for accessing Equinix Metal's API.
-func (c *Config) NewMetalGoClient() *metalv1.APIClient {
+func (c *Config) NewMetalClientForSDK(d *schema.ResourceData) *metalv1.APIClient {
+	client := c.newMetalClient()
```

Comment:
Note this means we're setting up a new client every time, which should avoid coordination issues for User-Agent modifications.  If we're OK with creating a bunch of separate clients, we could also use this pattern for other clients and avoid the occasional `concurrent map writes` panics we've seen (#286, #403).

Reply:
I don't. understand what the downside would be.

Reply:
I'm not totally clear on the overhead of creating these separate clients.  There's the potential for increased resource utilization, but we're still reusing the `http.DefaultTransport` across all clients so I _think_ any increase should be negligible.